### PR TITLE
Jufa/eb kernel states create ws

### PR DESCRIPTION
**Pull request by jfaltenbacher**
new article describing kernel state tasks for Elektrobit added and build environment created
Jira-ID:[ EB-324](https://jira.suse.com/browse/EB-324?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=1143140) 

please check language and structure
final PDF output attached
[SLE-RT-kernel-tasks_color_en_final.pdf](https://github.com/SUSE/doc-slert/files/7905082/SLE-RT-kernel-tasks_color_en_final.pdf)

### CODE REVIEW

File: xml/article_kernel_tasks.xml [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+  </listitem>
+ </varlistentry>
+ 
+ <varlistentry>
+  <term>
+   <parameter>TASK_PARKED</parameter>
+  </term>
+  <listitem>
+   <para>
+    A kernel thread is parked.
```
Comment (tomschr) [inline xml/article_kernel_tasks.xml:199]: Could it found a nice parking slot? :wink: Ok, seriously, I still don't know what that means. The parameter has "TASK" in its name, but you mention thread. I'm a bit confused.

**Review by tomschr — CHANGES_REQUESTED** [review]
@jfaltenbacher Thanks for your effort, much appreciated! :+1: 

I've left some smaller suggestions and some questions. Hope you find them helpful. If you have any questions let me know.

File: xml/article_kernel_tasks.xml [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+  xmlns:xlink="http://www.w3.org/1999/xlink">
+ <title>Processes and task states for Linux Kernel programming</title>
+ <info><productname>&productname;</productname>
+  <productnumber>&productnumber;</productnumber>
+  <xi:include href="common_copyright_gfdl.xml" parse="xml"/>
+  <date><?dbtimestamp?></date>
+  <abstract>
+   <para>
+    The following article describes the most important task states needed for
+    programming the Linux Kernel.
```
[the remark this answers is not in the document]
Comment (jfaltenbacher) [reply xml/article_kernel_tasks.xml:22]: the audience are experienced kernel developers; they should have profound knowledge about states and tasks and that is why there is not much explanation in the intro

File: xml/article_kernel_tasks.xml [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+   Each process can run through different states during its life cycle,
+   like <parameter>running</parameter> or <parameter>idle</parameter>.
+  </para>
+ 
+  <para>
+   A process might be unable to run (<parameter>idle</parameter>) at any time
+   and for different reasons.
+   For example, the process can be paused because it is waiting for input or
+   the scheduler has decided it has run the maximum amount of time units
+   allowed and that it must yield to another process.
```
[the remark this answers is not in the document]
Comment (jfaltenbacher) [reply xml/article_kernel_tasks.xml:45]: changed as suggested

File: xml/article_kernel_tasks.xml [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+  </para>
+
+<variablelist>
+ <varlistentry>
+  <term>
+   <parameter>TASK_NEW</parameter>
+  </term>
+  <listitem>
+   <para>
+    Marks a newly created task to prevent from external events (e.g. signals
```
[the remark this answers is not in the document]
Comment (jfaltenbacher) [reply xml/article_kernel_tasks.xml:60]: the styleguide only mentions "i.e." but I replaced it throughout the entire document. It also occurs in the diagram. Should I replace it there too? The texts will then be much longer in each box...

File: xml/article_kernel_tasks.xml [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+  </listitem>
+ </varlistentry>
+ 
+ <varlistentry>
+  <term>
+   <parameter>TASK_DEAD</parameter>
+  </term>
+  <listitem>
+   <para>
+    Task is ended but not deleted from the process table. It can be reported as
```
[the remark this answers is not in the document]
Comment (jfaltenbacher) [reply xml/article_kernel_tasks.xml:73]: in this case it is really ended and without errors. I can use _terminated_ but _stalled_ would not be correct.

File: xml/article_kernel_tasks.xml [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+  <term>
+   <parameter>TASK_UNINTERUPTIBLE</parameter>
+  </term>
+  <listitem>
+   <para>
+    Task is blocked while waiting for an event or condition, as for example
+    <parameter>mutex</parameter>. Any userspace interaction is deferred during
+    this time. Tasks blocked in this state, contribute to the load average. If
+    this tasks is blocked for too long, it can get reported by the hung task
+    detector.
```
[the remark this answers is not in the document]
Comment (jfaltenbacher) [reply xml/article_kernel_tasks.xml:122]: the point is, that some of these tasks can be reported by this detector some not. AFAIK from my discussion with our kernel expert, EB developers should be familiar with it

**Pull request merged**